plugins/win-wasapi/win-wasapi.cpp

```diff
@@ -1104,6 +1105,117 @@ static void *CreateWASAPISource(obs_data_t *settings, obs_source_t *source,
 	return nullptr;
 }
 
+typedef void(WINAPI *RtlGetVersion)(OSVERSIONINFOEXW *);
+bool isMediaCrashPatchNeeded()
+{
+	static bool first_attempt = true;
+	if (!first_attempt) {
+		return false;
+	}
+	first_attempt = false;
+
+	OSVERSIONINFOEXW osw = {0};
+	RtlGetVersion getVersion;
+	HMODULE ntdllmodule = LoadLibrary(TEXT("ntdll.dll"));
+	if (ntdllmodule) {
+		getVersion = (RtlGetVersion)GetProcAddress(ntdllmodule, "RtlGetVersion");
+		if (getVersion == 0) {
+			FreeLibrary(ntdllmodule);
+			return false;
+		}
+		osw.dwOSVersionInfoSize = sizeof(osw);
+		getVersion(&osw);
+		blog(LOG_DEBUG, "[MEDIADLLPATCH] windows version %d %d %d %d ",
+		     osw.dwBuildNumber, osw.dwMinorVersion, osw.dwMajorVersion, osw.dwPlatformId);
+		if (osw.dwBuildNumber == 22000 && osw.dwMinorVersion == 0 && osw.dwMajorVersion == 10) {
+			return true;
+		}
+	}
+	return false;
+}
+
+void patchMediaCrash()
+{
+	const uint8_t patterndata[] = {0x83, 0xF9, 0x08, 0xB8, 0x04, 0x00, 0x00,
```

Comment:
Maybe declare after sanity check to avoid allocating/deallocating unused arrays?

Reply:
updated.